### PR TITLE
Fix surface format for RGBA surfaces

### LINKED ISSUE #847: The new Colorspace implementation doesn't quite support RGBA, and that breaks jpg and png rendering
Both our JPG and PNG decoders set surface format as RGBA.  This breaks some validation apps (they use cocos2d), where the colors of the PNG and JPG images are off.

### PR DESCRIPTION
**Pull request by rajsesh-msft**
The correct surface format for Big Endian + Alpha Last is _ColorRGBA. We were incorrectly setting it to ARGB, resulting in apps that use this combination to get blue and red swapped and broke some validation apps.

To uncover future regressions, also modified one of the images to be PNG in WOCCatalog.  We still need additional validation through OpenGL samples.  Opened Issue #859 to track this.

Fixes #847

### CODE REVIEW

**Comment by rajsesh-msft** [comment]
@shahmeeresmail #Closed

**Comment by jaredhms** [comment]
>                         fmt = _ColorARGB;

are we able to verify the rest of these?  it seems like we should have a UT for _CGImageGetFormat for the various supported/unsupported formats.

---
Refers to: Frameworks/include/CGSurfaceInfoInternal.h:80 in a64e395. [](commit_id = a64e395cdb36e55afb50b118cffd8139dfdb7560, deletion_comment = False)

File: Frameworks/include/CGSurfaceInfoInternal.h [hunk 1]
```diff
@@ -91,8 +91,10 @@ static inline __CGSurfaceFormat _CGImageGetFormat(unsigned int bitsPerComponent,
                 case kCGImageAlphaNoneSkipLast:
```
Comment (rajsesh-msft) [inline Frameworks/include/CGSurfaceInfoInternal.h]: Should be RGBX.

**Comment by shahmeeresmail** [comment]
How is this hit in the current code path? I searched through the code base and even with the new changes, I could not find an instance where we set both kCGBitmapByteOrder32Big and kCGImageAlphaNoneSkipLast simultaneously as the bitmapInfo for an object. Perhaps it is set somewhere as a magic number?

**Comment by rajsesh-msft** [comment]
This was found in some internal apps we use for validation.  The apps use cocos2d.  The app creates CGBitmapContext with these surface spec, draws into it, and uses the bytes through openGL.  

---
In reply to: [238941568](https://github.com/Microsoft/WinObjC/pull/860#issuecomment-238941568) [](ancestors = 238941568)

**Comment by rajsesh-msft** [comment]
>                         fmt = _ColorARGB;

No.  validating this function is of no real use.  What we need really are true e2e validation for different apis that go through this.  #859 should really track it.

---
In reply to: [238881925](https://github.com/Microsoft/WinObjC/pull/860#issuecomment-238881925) [](ancestors = 238881925)

---
Refers to: Frameworks/include/CGSurfaceInfoInternal.h:80 in a64e395. [](commit_id = a64e395cdb36e55afb50b118cffd8139dfdb7560, deletion_comment = False)

**Comment by jaredhms** [comment]
:shipit:

**Comment by mnithish** [comment]
:shipit:

**Comment by bbowman** [comment]
:shipit:

**Pull request merged**